### ISSUE TITLE
Domain fronting support

### ISSUE THREAD

**Issue by ScriptIdiot** [MythicAgents/tetanus#4]
May I know if domain fronting is support in this case? Tried to change the host header but then it shows this error upon execution:

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: ConnectionRefused, error: "Failed to make post request" }', src/profiles/mod.rs:117:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

*Run on linux

Thanks!

**Comment by t94j0** [comment]
I haven't tried domain fronting with this agent, but I don't think there's an option to change the host header in the options of this agent. How did you attempt to change the host header?

**Comment by ScriptIdiot** [comment]
During the generation of payload, I was able to add host header to attempt to achieve domain fronting, which is working in merlin agent

**Comment by t94j0** [comment]
Gotcha. The Host header should be there based on [this code](https://github.com/MythicAgents/tetanus/blob/7ea02c90dbe0fb4dd200ef48e98d94f6eab59918/Payload_Type/tetanus/agent_code/src/profiles/http.rs#L117). It looks like you're getting a ConnectionRefused error, so it looks like the domain front you're connecting to is unavailable. Are you sure you can connect to the target domain/ip?

**Comment by ScriptIdiot** [comment]
Yeah, I am pretty sure. Tried to establish C2 without domain front, and the exact same setting with domain front worked by using merlin agent.

**Comment by MEhrn00** [comment]
I believe I have tracked down the issue. The HTTP library tetanus is using is minreq and minreq will insert the callback url into the Host header before establishing the HTTP connection. The library will not check for duplicates so it will include two Host headers but pick the Host specified in the HTTP Headers section of the payload generation for the HTTP connection and ignore the callback host.

This is obviously not great behavior at all and I will need to use a different library which supports arbitrarily setting the Host header. I will look into it this weekend.

**Comment by jeffmcjunkin** [comment]
For what it's worth, between this and #1 tetanus is getting damn close to a viable top-tier C2. The only ~two~ three wish list items I'd have left are the following:

1. .NET Assembly injection support
2. Beacon Object File support (likely via https://github.com/trustedsec/COFFLoader)
3. DNS C2 channel support

The first two get a lot of existing functionality for free (via projects like https://github.com/trustedsec/CS-Situational-Awareness-BOF and https://github.com/Flangvik/SharpCollection), while the third makes operations possible in a pretty restricted environment.

**Comment by jeffmcjunkin** [comment]
Oooh, today I learned of https://github.com/hakaioffsec/coffee, which may be of interest as an easier path to implementing Beacon Object Files. 

In other news, is there any rough planned timeframe for the rewrite / release?
